lighthouse-cli/bin.ts

```diff
@@ -271,24 +274,25 @@ function runLighthouse(url: string,
       return results;
     })
     .then((results: Results) => {
-      results.artifacts = {};
+      // remove artifacts so users won't get unnecessary artifacts in the report.
+      results.artifacts = undefined;
       return results;
      })
     .then((results: Results) => Printer.write(results, flags.output, flags.outputPath))
     .then((results: Results) => {
-      const filename = `${assetSaver.getFilenamePrefix({url})}.report.html`;
-
       if (flags.output === Printer.OutputMode[Printer.OutputMode.pretty]) {
-        Printer.write(results, 'html', filename);
+        const filename = `${assetSaver.getFilenamePrefix({url})}.report.html`;
+        return Printer.write(results, 'html', filename);
       }
-
+      return results;
+    })
+    .then((results: Results) => {
       // If --view, generate report.html, host it, and open it in the default browser
       if (flags.view) {
-        return performanceXServer.startServer(0).then((port: number) => {
-          const filePath = `${performanceXServer.FOLDERS.REPORTS}/${filename}`;
-          return Printer.write(results, 'html', filePath).then(() => {
-            opn(`http://localhost:${port}/reports/${filename}`);
-          });
+        const filename = `${assetSaver.getFilenamePrefix({url})}.report.html`;
```

Comment:
Although this line can be combined with line 284, we need a variable outside of the 'then-chain' to do that. Also, since we no longer use symlinks, the filenames do not need to be the same. And in the future, we may save JSON instead of html.